lib/util/hud_code_gen.rb

```diff
@@ -141,4 +142,108 @@ module Types::HmisSchema::Enums::Hud
 
     [map_name, lookup_fn_name]
   end
+
+  def generate_hud_list_json(year, excel_file_path)
+    # Parse the existing JSON data. This will be the base from which we start.
+    json_file_path = "lib/data/#{year}_hud_lists.json"
+    json_data = JSON.parse(File.read(json_file_path))
+
+    # TODO: is there a way to better skip these?
+    ignore_codes = ['List', '*', '(see note)']
+    excel_file = Roo::Excelx.new(excel_file_path)
+
+    # Parse Excel file for list codes
+    excel_data = {}.tap do |codes|
+      # Sheet 0: CSV|DE#|Name|Type|List|Null|Notes|Validate
+      # Pull from this sheet to get the names for each list of codes
+      excel_file.sheet(0).each(code: 'List', name: 'Name') do |row|
+        # Clean data coming in from excel
+        code = clean_json_text(row[:code].to_s)
+        list_name = clean_json_text(row[:name].to_s)
+
+        # Only need rows that have referenced codes - not all data on this sheet utilizes a code list
+        unless code.blank? || ignore_codes.include?(code)
+          codes[code] ||= {}
+          (codes[code][:list_name] ||= []) << list_name
+        end
+      end
+      # Sheet 1: List|Value|Text
+      # This sheet does not have names for the lists, but it may include some that are note referenced in
+      # the first sheet. In this case, bring the code list in with a name "Unknown". These ones will need
+      # to be manually checked and named.
+      excel_file.sheet(1).each(code: 'List', value: 'Value', text: 'Text') do |row|
+        # Clean data coming in from excel
+        code = clean_json_text(row[:code].to_s)
+        value = clean_json_text(row[:value].to_s)
+        text = clean_json_text(row[:text].to_s)
+        value = value.to_i if Integer(value, exception: false)
+
+        # Skip header and blank rows if they exist
+        unless code.blank? || ignore_codes.include?(code)
+          codes[code] ||= { list_name: ['Unknown'] }
+          (codes[code][:values] ||= {})[value] = text
```

Comment:
I'm probably missing something, but ?
```suggestion
          codes[code] ||= { list_name: ['Unknown'], values: {} }
          codes[code][:values][value] = text
```